live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/ApplicationInfoProvider.java

```diff
@@ -0,0 +1,63 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.adapters.messaging.repository;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.logging.log4j.util.Strings;
+import org.springframework.stereotype.Component;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.IntegrationProperties;
+import org.alfresco.hxi_connector.live_ingester.domain.ports.repository.api.DiscoveryApi;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class ApplicationInfoProvider
+{
+
+    private final DiscoveryApi discoveryApi;
+    private final IntegrationProperties integrationProperties;
+
+    private String applicationInfo;
+
+    public String getUserAgentData()
+    {
+        if (Strings.isBlank(applicationInfo))
+        {
+            applicationInfo = calculateUserAgentData();
+        }
+        return applicationInfo;
+    }
+
+    private String calculateUserAgentData()
+    {
+        String applicationVersion = integrationProperties.application().version();
+        String repositoryVersion = discoveryApi.getRepositoryVersion();
```

Comment:
I assume that we totally give up on the "running bulk ingester without repo" idea

Reply:
Great point Pawel. I'll create an extra ticket so that we can alternatively get the repository version from configuration.

Reply:
The new ticket is here: https://hyland.atlassian.net/browse/ACS-8423

Reply:
Thanks for catching that Paweł.